src/error.rs

```diff
@@ -113,24 +113,154 @@ pub enum CmdError {
 }
 
 impl CmdError {
-    /// Returns true if this error indicates that a matching element was not found.
-    ///
-    /// Equivalent to
-    /// ```no_run
-    /// # use fantoccini::error::{CmdError, ErrorStatus};
-    /// # let e = CmdError::NotJson(String::new());
-    /// let is_miss = if let CmdError::Standard(w) = e {
-    ///     matches!(w.error, ErrorStatus::NoSuchElement)
-    /// } else {
-    ///     false
-    /// };
-    /// ```
-    pub fn is_miss(&self) -> bool {
-        if let CmdError::Standard(w) = self {
-            matches!(w.error, ErrorStatus::NoSuchElement)
-        } else {
-            false
-        }
+    /// Return true if this error matches [`ErrorStatus::DetachedShadowRoot`].
+    pub fn is_detached_shadow_root(&self) -> bool {
+        matches!(self, CmdError::Standard(w) if w.error == ErrorStatus::DetachedShadowRoot)
+    }
```

Comment:
Not a blocker, but it'd be nice if we could turn all of these nearly-identical definitions into a macro. Something like:
```rust
macro_rules! is_helper {
    ($($variant:ident => $name:ident$(,)?),*) => {
        $(
            /// Return true if this error matches
            #[doc = concat!("[`ErrorStatus::", stringify!($variant), "`].")]
            pub fn $name(&self) -> bool {
                matches!(self, CmdError::Standard(w) if w.error == ErrorStatus::$variant)
            }
        )*
    }
}
```
You can then replace all of the methods with something like:
```rust
impl CmdError {
    is_helper! {
        Foo => is_foo,
        Bar => is_bar,
    }
}
```

Reply:
Good idea. I actually watched your livestream of this 😄 This macro worked a treat. Thanks!